### PR TITLE
python3.12 support

### PR DESCRIPTION
**Pull request by chidanandpujar**
Python3.12 support
#1253
#1252 
#1280
#1276

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 lxml>=3.2.4
 # ncclient version 0.6.10 has issues with PyEZ(junos-eznc) and needs to be avoided
-ncclient==0.6.13
+ncclient==0.6.15
```
Comment (ktbyers) [inline requirements.txt:3]: Really should make this a `ncclient>=0.6.15` (to try avoid pinning to a specific version of ncclient).

Comment (chidanandpujar) [reply requirements.txt:3]: Hi @ktbyers 
    Thanks vey much for the review comments .
   I will update the fix .

Thanks

**Comment by ktbyers** [comment]
We should update this (so newer versions of Python have some testing in the CI/CD system):

https://github.com/Juniper/py-junos-eznc/blob/master/.github/workflows/pylint.yml#L17

@chidanandpujar Do you want to add that into this PR?

**Comment by chidanandpujar** [comment]
> We should update this (so newer versions of Python have some testing in the CI/CD system):
> 
> https://github.com/Juniper/py-junos-eznc/blob/master/.github/workflows/pylint.yml#L17
> 
> @chidanandpujar Do you want to add that into this PR?

Hi @ktbyers 
    Thanks for the review comments .
    We will update the github workflow with never versions on python.

Thanks

**Pull request closed without merging**